main.go

```diff
@@ -50,38 +47,28 @@ func init() {
 func main() {
 	ctrl.SetLogger(zap.New(zap.UseDevMode(debug)))
 
-	setupLog.Info(version.ToString())
-
-	nonCached := []client.Object{
-		&wfv1.Workflow{},
-		&addonmgrv1alpha1.Addon{},
-		&apiextensionsv1.CustomResourceDefinition{},
-	}
-	mgr, err := ctrl.NewManager(ctrl.GetConfigOrDie(), ctrl.Options{
```

Comment:
I don't agree w/ this change, moving away from `controller-runtime` manager puts this entire controller onto a custom path outside of recommended (opionated) way of kubebuilder framework, I also do not see good reason for it as you'll end up re-implementing parts of the runtime and apimachinery do to this drift.

Why are we rewriting manager?

Reply:
We do not re-write manager at all - we bump controller Run() directly instead using a wrapper the manager. 
Still, we need kubebuilder builds RBAC(s) stuff so that controller is able to watch and list.

Reply:
We should put the controller.Run into a cobra.Command.